Update the documentation style (#148)
Babylonpartners/ios-playbook: Cookbook/Technical-Documents/FastlaneMatch.md

Fastlane Match
==============

We use [Fastlane Match](https://docs.fastlane.tools/actions/match/) to share code-signing identities across the team and CI. We have configured it to store the identities in our private git repo

**Caution**

- Use Match with `--readonly false` **only if** you need to generate/regenerate profiles. Otherwise existing builds on Hockeyapp will become invalid. By default it is run with `--readonly true`
- **DO NOT** use Xcode code signing capabilities ("Download Manual Profiles" or "Manage Certificates" buttons in Xcode's preferences pane) and don't change automatic code signing in the project settings. Always use Match.

## Download and install code signing identities on your machine

1. Add shared developer account credentials from 1Password to your Xcode settings in `Preferences -> Accounts` menu. 

> 2. Run 

Author: :D Let me fix that

```shell
 $ bundle exec fastlane match {{Type}} --team_id {{Team-ID}} 
```

> Note:- 

Reviewer: ```suggestion
Note:
```

> * We have 2 developer teams - Enterprise and App-store 

Author: I did not want to mention app-store team there because these are handled specially as described in next section. I did not want a developer to read first section and run the command with just a couple of options

> * We have 2 developer teams - Enterprise and App-store 

Reviewer 2: Ok, make sense, but then I think the wording is confusing.

Maybe find a place upstream, like a paragraph used as introduction of them all, to say:

> We use two teams on our portal:
> - one "Enterprise" team we use for Day to Day developent and HockeyApp (see <link to paragraph>)
> - one "AppStore" team used for TestFlight and AppStore (see <link to paragraph>)

Then in the specific paragraphs, just reference them, e.g.:
 - remove the "We have two developer teams – Enterprise and App-store" line from here (since you'd have moved it upstream in the intro) but instead say "For day to day and HockeyApp, we use certificates and profiles from our Development team"
 - in the paragraph about TestFlight and AppStore, put a similar paragraph like "For TestFlight and AppStore, we use certificates and profiles from our App Store developer team"

* Following options can be used for the command above:

| Code signing identities for | Type param| Team-ID param    |
|-----------------------------|-----------|-------------------
|Day-to-day development       |development|Enterprise team ID|
|Hockeyapp                    |enterprise |Enterprise team ID|

3. On prompt for password to decrypt the repository, enter the one from 1Password vault.

To troubleshoot, refer to `fastlane/Matchfile`, `fastlane/Appfile` and [documentation](https://docs.fastlane.tools/actions/match/)

### TestFlight & AppStore

To install certificates and provisioning profiles for creating a Testflight or App Store build locally, run the command

```shell

>  $ bundle exec fastlane match appstore --team_id <appstore team id> --git_branch <appstore team id> --app_identifier <appstore bundle id>

Reviewer 2: I'd rather use the same placeholder names and syntax throughout the whole document. So replacing `<appstore team id>` with `((Team-ID}}` (nit: I even prefer `{{TeamID}}` without the dash), and use always the same delimiters (either curly braces or angle brackets but not a mix of both)

Author: Good observation. I had left the previous commands pretty much untouched and for the ones I wrote, I used curly braces

```

Note: 
- More on `--git-branch` option [below](#several-targets-and-teams)

> - The reason to specify individual app identifiers is that otherwise if AppStore bundle ids are added to the default set of identifiers Match will try to load certificates that may not exist for some apps on the portal, as those apps are registered in different teams.

Reviewer 3: ```
The reason to specify individual app identifiers is that otherwise if AppStore bundle ids are added to the default set of identifiers Match will try to load certificates that may not exist for some apps on the portal, as those apps are registered in different teams.
```
->
```
We specify individual app identifiers instead of adding all App Store bundle ids to the default set as Match would otherwise attempt to load certificates that do not exist in the portal. This is because some of our app ids are registered with teams that belong to our partners.
```

Author: Nice. I was struggling to phrase it nicely myself :)

> - The reason to specify individual app identifiers is that otherwise if AppStore bundle ids are added to the default set of identifiers Match will try to load certificates that may not exist for some apps on the portal, as those apps are registered in different teams.

Author: ```suggestion
We specify individual app identifiers instead of adding all App Store bundle ids to the default set as Match would otherwise attempt to load certificates that do not exist in the portal. This is because some of our app ids are registered with teams that belong to our partners.
```

## Renew expired profiles

> (At the time of writing this, `fastlane` could not renew expired profiles automatically or using a command although the documentation mentions that it does)

Reviewer 3: Please add the date here. Otherwise I will not know whether it is worth checking if the fastlane has learned how to renew expired profiles.

- Renew the profile on Apple developer center
- Run the following command 

```shell
 $ bundle exec fastlane match appstore --team_id <appstore team id> --git_branch <appstore team id> --app_identifier <appstore bundle id> --readonly false
```
Note: We specify `app-identifier` so that other valid profiles are not re-generated by mistake

## Add new device for development

> - Register device manually on the portal in enterpirse team account

Reviewer 3: ```suggestion
- Register device manually on the portal in the enterprise team account
```
